### ISSUE TITLE
Poor buffering behavior when not transcoding files on cloud storage

### ISSUE THREAD

**Issue by fuzxi** [iwalton3/jellyfin-mpv-shim#22]
**Describe the bug**
This seems like the opposite of expected behavior with regard to the increased resource demands of transcoding, but I experience horrible buffering only when direct-playing files stored on my Google Drive with jellyfin-mpv-shim. 
This does not occur with files stored locally on my home server. 
This does not occur when transcoding files, regardless of where they are stored.
This does not occur when using other clients, including the Jellyfin web client and the Jellyfin Android app.

**To Reproduce**
1. Play any file stored on a Google Drive rclone mount.
2. Observe buffer.

The file will open and buffer 1-10 seconds of playback. The buffer will slowly decrease in size over the next 10-20 seconds until it reaches 0 and playback stutters indefinitely.

**Expected behavior**

The file should open and buffer ~30 seconds of playback, then continue to fill up the buffer to several minutes of playback. The buffer should remain full or nearly full for the duration of playback.

**Desktop (please complete the following information):**
 - OS: Manjaro Linux
 - Version 1.3.10

**Error Messages**

_1080p HI-10p file, 1 minute of playback_

No transcoding. Buffer reached 0 seconds 5 times, buffer was empty at 1 minute point.
[shim1.txt](https://github.com/iwalton3/jellyfin-mpv-shim/files/4233895/shim1.txt)
Transcoding, video quality set to "Maximum". Buffer never reached 0 seconds, buffer had 2:01 cached at 1 minute point.
[shim2.txt](https://github.com/iwalton3/jellyfin-mpv-shim/files/4233897/shim2.txt)

_1080p yuv420 (e.g. not HI-10p) file, 30 seconds of playback_

No transcoding. Buffer reached 0 seconds too many times to count, buffer was empty at 30 second point.
[shim3.txt](https://github.com/iwalton3/jellyfin-mpv-shim/files/4233901/shim3.txt)
Transcoding, video quality set to "Maximum". Buffer never reached 0 seconds, buffer had 1:20 cached at 30 second point.
[shim4.txt](https://github.com/iwalton3/jellyfin-mpv-shim/files/4233903/shim4.txt)

I don't have any x265 files to test.

**Comment by iwalton3** [comment]
MPV Shim doesn't keep any logs from MPV itself, so unfortunately those logs don't tell me much about the stream quality issue. The player sends the direct and HLS transcode URLs directly to MPV. I wonder if this has something to do with the way that Jellyfin serves direct URLs.

You can configure the [MPV cache](https://mpv.io/manual/stable/#cache) using `mpv.conf` in the config directory, which is in `~/.config/jellyfin-mpv-shim`. I would suggest trying various cache settings, including no cache, a small cache (like a few seconds), a medium cache (a few minutes), and a very large cache (as much RAM as you want to give to MPV).

If you want to see log messages from MPV, the easiest way is to configure an external MPV. Specify the following config options in `conf.json`:
```
"mpv_ext": true,
"mpv_ext_path": null,
"mpv_ext_ipc": "/tmp/mpv-socket",
"mpv_ext_start": false,
```
Then launch mpv with `mpv -v --input-ipc-server=/tmp/mpv-socket --idle` and launch the shim. Any errors from MPV should show in the console where you started it.

What could be helpful is an I/O graph of Rclone and Jellyfin logs. I do also have support for direct paths in MPV Shim, but as far as I'm aware that doesn't work on the current stable JF version.

**Comment by iwalton3** [comment]
It is now much easier to log the output from MPV. The latest version of the software logs basic MPV output by default, and you can increase the log level in the config file. (I recommend using `debug` as the log level if you are troubleshooting MPV.)

**Comment by fuzxi** [comment]
After a lot of tinkering, I made 2 changes which finally resolved this issue. I changed my rclone mount setting of `vfs-cache-mode` from `minimal` to `off`, and replaced my unionfs mount with an rclone union mount after realizing that the unionfs mount had performance issues. 

All files now buffer and play as expected, without interruption. I'm still mystified as to why I was getting better performance when transcoding than when direct playing, and why it wasn't an issue in the web player, but problem solved.
